BedrockCommand.cpp

```diff
@@ -127,6 +130,83 @@ bool BedrockCommand::areHttpsRequestsComplete() const {
     return true;
 }
 
+void BedrockCommand::_waitForHTTPSRequests() {
+    uint64_t startTime = 0;
+    while (!areHttpsRequestsComplete()) {
+        // Wait until the command's timeout, or break early if the command has timed out.
+        uint64_t maxWaitUs = 0;
+        uint64_t now = STimeNow();
+        if (!startTime) {
+            startTime = now;
+        }
+        if (now < timeout()) {
+            maxWaitUs = timeout() - now;
+        } else {
+            // This uses the same starting point as in areHttpsRequestsComplete, for efficiency.
+            // We won't iterate over large numbers of known completed requests, instead starting
+            // from the the point of the list of known completed request.
+            auto requestIt = (_lastContiguousCompletedTransaction == httpsRequests.end()) ? httpsRequests.begin() : _lastContiguousCompletedTransaction;
+            while (requestIt != httpsRequests.end()) {
+                if (!(*requestIt)->response) {
+                    (*requestIt)->response = 500;
+                }
+                requestIt++;
+            }
+
+            // Timed everything out, can return.
+            break;
+        }
+
+        fd_map fdm;
+        prePoll(fdm);
+
+        // We never wait more than 1 second in `poll`. There are two uses for this. One is that at shutdown, we want to kill any sockets that have are making no progress.
+        // We don't want these to be stuck sitting for 5 minutes doing nothing while thew server hangs, so we will interrupt every second to check on them.
+        // The other case is that there can be no sockets at all.
+        // Why would there be no sockets? It's because Auth::Stripe, as a rate-limiting feature, attaches sockets to requests after their made.
```

Comment:
```suggestion
        // Why would there be no sockets? It's because Auth::Stripe, as a rate-limiting feature, attaches sockets to requests after they're made.
```